packages/design-system/src/new-ui/Button/Button.stories.tsx

```diff
@@ -100,3 +101,35 @@ export const WithRoundedNone: Story = {
     </Button>
   ),
 };
+
+export const SortingButton = () => {
+  const [sorting, setSorting] = React.useState("");
+
+  const toggleSorting = (type: string) => {
+    if (type === "asc") {
+      setSorting("desc");
+    } else {
+      setSorting("asc");
+    }
+  };
+
+  return (
+    <Button
+      className="gap-x-2"
+      variant="tertiaryGrey"
+      size="sm"
+      onClick={() => toggleSorting(sorting)}
+    >
+      Sorting
+      {!sorting && (
+        <Icons.ChevronSelectorVertical className="h-4 w-4 stroke-semantic-fg-secondary" />
+      )}
+      {sorting === "asc" && (
+        <Icons.ArrowDown className="h-4 w-4 stroke-semantic-fg-secondary" />
+      )}
+      {sorting === "desc" && (
+        <Icons.ArrowUp className="h-4 w-4 stroke-semantic-fg-secondary" />
+      )}
+    </Button>
+  );
+};
```

Comment:
@iamnamananand996 This has a issue. The sorting has only three state "asc" | "desc" | null. In this case you have to make the type explicit not just string. Besides that, please don't use the short-hand operator on UI. Use ternary operator instead. 

Please change this accordingly